runtime-light/stdlib/time/time-functions.h

```diff
@@ -68,34 +78,83 @@ inline int64_t f$time() noexcept {
   return duration_cast<chrono::seconds>(chrono::system_clock::now().time_since_epoch()).count();
 }
 
-inline int64_t f$mktime(Optional<int64_t> hour = {}, Optional<int64_t> minute = {}, Optional<int64_t> second = {}, Optional<int64_t> month = {},
-                        Optional<int64_t> day = {}, Optional<int64_t> year = {}) noexcept {
-  namespace chrono = std::chrono;
-  const auto time_since_epoch{chrono::system_clock::now().time_since_epoch()};
-  chrono::year_month_day current_date{chrono::sys_days{duration_cast<chrono::days>(time_since_epoch)}};
+inline int64_t f$mktime(int64_t hour = std::numeric_limits<int64_t>::min(), int64_t minute = std::numeric_limits<int64_t>::min(),
+                        int64_t second = std::numeric_limits<int64_t>::min(), int64_t month = std::numeric_limits<int64_t>::min(),
+                        int64_t day = std::numeric_limits<int64_t>::min(), int64_t year = std::numeric_limits<int64_t>::min()) noexcept {
+  auto res{kphp::timelib::mktime(hour != std::numeric_limits<int64_t>::min() ? std::make_optional(hour) : std::nullopt,
+                                 minute != std::numeric_limits<int64_t>::min() ? std::make_optional(minute) : std::nullopt,
+                                 second != std::numeric_limits<int64_t>::min() ? std::make_optional(second) : std::nullopt,
+                                 month != std::numeric_limits<int64_t>::min() ? std::make_optional(month) : std::nullopt,
+                                 day != std::numeric_limits<int64_t>::min() ? std::make_optional(day) : std::nullopt,
+                                 year != std::numeric_limits<int64_t>::min() ? std::make_optional(kphp::time::impl::fix_year(year)) : std::nullopt)};
+  if (res.has_value()) {
+    return *res;
+  }
+  return std::numeric_limits<int64_t>::min();
+}
+
+inline array<mixed> f$getdate(int64_t timestamp) noexcept {
+  if (timestamp == std::numeric_limits<int64_t>::min()) {
+    namespace chrono = std::chrono;
+    timestamp = static_cast<int64_t>(chrono::time_point_cast<chrono::seconds>(chrono::system_clock::now()).time_since_epoch().count());
+  }
+  tm t{};
+  time_t timestamp_t{timestamp};
+  tm* tp{k2::localtime_r(std::addressof(timestamp_t), std::addressof(t))};
+  if (tp == nullptr) {
+    std::array<char, 256> buff{};
+    kphp::log::warning("Error \"{}\" in getdate with timestamp {}", strerror_r(errno, buff.data(), buff.size()), timestamp);
```

Comment:
- nit: we prefer logs starting with lower case letters
- As I see, `k2_localtime_r` doesn't state that it sets `errno`